### ISSUE TITLE
[Bug]: Viewports aren’t reset when navigating away from a story with a default viewport

### ISSUE THREAD

**Issue by JReinhold** [storybookjs/storybook#27073]
### Describe the bug

When navigating to a story with a `defaultViewport`, that viewport is correctly applied, and the `globals` query param in the URL is updated accordingly. When navigating away to a story without an explicit viewport, the global stays there, meaning that the viewport is still applied, even though the story doesn't have a default viewport.

### To Reproduce

- Open a regular story without a viewport: [https://next--635781f3500dd2c49e189caf.chromatic.com/?path=/story/components-modal--fixed-height](https://next--635781f3500dd2c49e189caf.chromatic.com/?path=/story/components-modal--fixed-height&globals=viewport:mobile1)
- Navigate to the story Manager/Mobile/About/Default - which has a default mobile viewport
- Navigate to a story without a viewport like Components/Modal/Default
- See that the mobile viewport is still applied even though it’s not part of that story

### System

```bash
Storybook 8.1.0-alpha.8
```

**Comment by shilman** [comment]
I updated the viewports addon to store its state in globals in 8.0 as a step towards standardizing on globals for most addon state. This results in the behavior you describe and I don't plan to fix this.

@tmeasday is working on global overrides per https://github.com/storybookjs/storybook/discussions/23347 and if those are finished it would have the desired behavior you describe when completed.

**Comment by DanielFlores-OG** [comment]
I have the same issue. Pretty annoying...


A workaround is to set disable: true in preview.js

` paremeters{viewport: { viewports, disable: true }}`

and then disable: false in every story (or Meta) where you want to apply some viewport:

 `   parameters: {
    viewport: {
      disable: false,
      defaultViewport: "s",
    },
  },
`

**Comment by jpzwarte** [comment]
One workaround is to explicitly define the `defaultViewport` in the default export: 

```
export default {
  title: '...,
  parameters: {
    viewport: {
      defaultViewport: 'default'
    }
  }
};
```

**Comment by lundmikkel** [comment]
@jpzwarte's fix seems to work with `preview.ts` as well:

```ts
const preview: Preview = {
  parameters: {
    viewport: {
      defaultViewport: "reset",
    },
  },
};
export default preview;
```

**Comment by jpzwarte** [comment]
@lundmikkel i *think* my workaround is obsolete with the release of Storybook 8.3 where you can now specify globals, right @JReinhold ?

![Image-1](https://github.com/user-attachments/assets/79cd4f52-ff72-433b-a61e-a53d6c05f689)
(copied from a post on X)

**Comment by JReinhold** [comment]
Yes I think if you migrate to the new globals syntax for viewports, that should also fix the issue.

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-parameters-format-for-addon-viewport

**Comment by vanessayuenn** [comment]
Closing this as we have now a solution!

**Comment by TomasSestak** [comment]
I think this issue should be reopened because @jpzwarte's solution stopped working with the new release and adding this to preview.tsx:

```
  initialGlobals: {
    viewport: { value: 'reset' }
  },
```

has no effect - stories now keep the previously selected viewport
